Task 1

enginire4ek opened the pull request:
Creating home screens
Navigating with Navigation component

enginire4ek committed:
Task 1
Creating home screens
Navigating with Navigation component

enginire4ek committed:
Linter fixes
- Deleted checkstyle from pr.yml
- Updated Android Gradle plugin to Java 11

enginire4ek committed:
Linter fixes
- Updated JAVA 11 correctly

enginire4ek committed:
Linter fixes
- Changed java.home path

enginire4ek committed:
Linter fixes
- Changed JAVA version

enginire4ek committed:
A few moments later...

enginire4ek committed:
Last change

102Rudy commented on app/src/main/java/com/example/pinterbest/HomeFragment.kt:
```diff
@@ -0,0 +1,74 @@
+import com.example.pinterbest.databinding.FragmentHomeBinding
+
+class HomeFragment : Fragment() {
+    private var _binding: FragmentHomeBinding? = null
+    private val binding get() = _binding!!
```
take a look on `lateinit var`, it's great approach to avoid force unwrap `!!` and keep clean solution in this case

enginire4ek replied:
Fixed here: https://github.com/enginire4ek/Pinterbest/pull/1/commits/c349b66d96a4b6bbf03557b4bc3771de1638810b

102Rudy commented on app/src/main/java/com/example/pinterbest/HomeFragment.kt:
```diff
@@ -0,0 +1,74 @@
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
```
it's unnecessarily since `_binding` is used only in this class

102Rudy commented on app/src/main/java/com/example/pinterbest/LoginFragment.kt:
```diff
@@ -0,0 +1,61 @@
+                it.findNavController().navigate(R.id.action_loginFragment_to_homeFragment)
+            } else {
+                Toast.makeText(context, "Not correct", Toast.LENGTH_SHORT).show()
+                // Для облегчения входа в приложение
```
nit: the language of development is English, so let's use it even in comments

enginire4ek replied:
Fixed here: https://github.com/enginire4ek/Pinterbest/pull/1/commits/c349b66d96a4b6bbf03557b4bc3771de1638810b

102Rudy commented on app/src/main/java/com/example/pinterbest/MainActivity.kt:
```diff
@@ -0,0 +1,105 @@
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Устанавливаем BottomNavigationBar
```
looks like useless comment. Overall, try to stick to approach of self documenting code

enginire4ek replied:
Fixed here: https://github.com/enginire4ek/Pinterbest/pull/1/commits/c349b66d96a4b6bbf03557b4bc3771de1638810b

102Rudy commented on app/src/main/java/com/example/pinterbest/adapters/PinFeedHomeAdapter.kt:
```diff
@@ -0,0 +1,38 @@
+class PinFeedHomeAdapter(private val pinObjects: List<PinObject>) :
+    RecyclerView.Adapter<PinFeedHomeAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.view_holder_home_feed, parent, false)
+        return ViewHolder(view)
+    }
```
If you use view binding in Fragments, let's use it in Adapters & ViewHolders too

102Rudy commented on app/src/main/java/com/example/pinterbest/utilities/Validator.kt:
```diff
@@ -0,0 +1,179 @@
+    companion object {
+
+        // Default validation messages
+        private val PASSWORD_POLICY =
+            """Password should be minimum 8 characters long,
+            |should contain at least one capital letter,
+            |at least one small letter,
+            |at least one number and
+            |at least one special character among ~!@#$%^&*()-_=+|[]{};:'\",<.>/?""".trimMargin()
+
+        private const val NAME_VALIDATION_MSG = "Enter a valid name"
+        private const val EMAIL_VALIDATION_MSG = "Enter a valid email address"
+        private const val PHONE_VALIDATION_MSG = "Enter a valid phone number"
```
This and others texts which are showed to user must be placed in string resources files

enginire4ek replied:
Fixed here: https://github.com/enginire4ek/Pinterbest/pull/1/commits/c349b66d96a4b6bbf03557b4bc3771de1638810b

102Rudy commented on app/src/main/java/com/example/pinterbest/utilities/Validator.kt:
```diff
@@ -0,0 +1,179 @@
+import com.google.android.material.textfield.TextInputLayout
+import java.util.regex.Pattern
+
+class Validator private constructor() {
+    companion object {
```
you can use `object Validator {` to make the same behavior

enginire4ek replied:
Fixed here: https://github.com/enginire4ek/Pinterbest/pull/1/commits/c349b66d96a4b6bbf03557b4bc3771de1638810b

Commit:
Task 1 (changes)
- binding variable declaration changed in all fragments
- deleted dataBinding
- Changed root element of all layouts

enginire4ek committed:
Task 1(changes)
- Add viewBinding in PinFeedHomeAdapter class
